canvas_modules/harness/src/client/App.js

```diff
@@ -1469,6 +1469,8 @@ class App extends React.Component {
 				iconComponenet = null;
 			}
 			callbackIcon(iconComponenet);
+		} else if (data.type === "customDataTypeIcon") { // custom icon from consumer app for custom data type
```

Comment:
Can you add `dataType` to this example so that a different icon is returned
<img width="251" height="48" alt="image" src="https://github.com/user-attachments/assets/dbb1d7cb-ac1e-4d5b-8e01-0fe6ca0b8e1e" />